packages/scrollable-image/src/components/lazy-image.js

```diff
@@ -0,0 +1,75 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import styled from 'styled-components'
+
+const Placeholder = styled.div`
+  width: 100vw;
+  height: 100vh;
+`
+
+const Container = styled.div`
+  display: inline-block;
+  img {
+    width: auto;
+    height: 100vh;
+  }
+`
+
+class LazyImage extends React.PureComponent {
+  static propTypes = {
+    src: PropTypes.string.isRequired,
+    onLoad: PropTypes.func.isRequired,
+    onError: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      readyToLoad: false,
+    }
+    this.handleLoad = this._handleLoad.bind(this)
+    this.setReady = this._setReady.bind(this)
+  }
+
+  componentDidMount() {
+    this.isWindowLoad(this.setReady)
+  }
+
+  componentWillUnMount() {
+    window.removeEventListener('load', this.setReady)
+  }
+
+  isWindowLoad(callback) {
+    if (document.readyState === 'complete') {
```

Comment:
if we want to lazy load the images after window is loaded,
`document.readyState` might not be appropriate here.
`document.readyState` only indicates DOM is ready, all resources such as images, stylesheets might not be ready even if `document.readyState === 'complete'` .

Reply:
After searching a bit for [`document.readyState`](https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState) and [other events](https://developer.mozilla.org/en-US/docs/Web/API/Document/readystatechange_event) fired by document, here's what I have learned: 

When `document.readyState === 'complete'`, the document and all sub-resources (stylesheets, images) have finished loading and `load` event is about to fire.  To my understanding, this state means document is ready.

Function `isWindowLoad` here check `document.readyState === 'complete'` for document which already fired `load` event before I add an event listener for it.

Reply:
Oops, my bad, I misunderstand `document.readyState`.